packages/esm-patient-registration-app/src/patient-registration/field/obs/obs-field.component.tsx

```diff
@@ -159,14 +161,43 @@ interface DateObsFieldProps {
   concept: ConceptResponse;
   label: string;
   required?: boolean;
-  dateFormat?: string;
-  placeholder?: string;
+  minDate?: string;
+  maxDate?: string;
 }
 
-function DateObsField({ concept, label, required, placeholder }: DateObsFieldProps) {
+const evaluateConfigDate = (dateString: string, dateType: string): Date | null => {
```

Comment:
So basically, the real feature being supported by this PR is the ability to specify `today` as either the lowest selectable date or the highest selectable date?

Reply:
Basically, the question here is "why don't we just build that instead of this?"

Reply:
+1, I would support a config interface like, for example,

```
datepicker: {
  allowPastDates: boolean
  allowFutureDates: boolean
}
```

does what's actually needed, is easier to maintain and code, and easier to understand and use.

Reply:
@ibacher & @brandones, my question here would be, don't we anticipate a scenario where someone would pass a fixed date e.g 2024-10-01 as either the `min` or `max` date. I don't have a use case per se but I just thought it would be possible.

Reply:
Yeah I just can't really imagine a use case for that. It is good not to build things no one needs, or to design and build things with no use case. It is valuable to keep our code easy to understand, maintain, and use. If someday a use case comes up, we'll implement it then.

Reply:
@ibacher / @brandones / @kajambiya please keep in mind that "today" from the perspective of a field on a form often needs to be "as of the encounter date" _not_ "as of the entry date".  So it is not sufficient to say we are only going to support past and future as of a "new Date()" object.  So this might be an example of where a parent component (eg. a parent form with an encounterDate variable would want to set that as the minDate / maxDate / defaultDate on an obsField it contains).  Sorry if this is not relevant, but wanted to mention it in case it is.

Reply:
@mseaton Yeah, retrospective registration probably requires a redesign of the app. Here, though, we're just passing in a value from the configuration, so the issue is kind of orthogonal to that concern (i.e., we could implement this so that it interprets "today" relative to the user's current selected time for the registration to happen). The issue here is that there's isn't a lot of use for handling strings like '2020-01-05'.